### PR TITLE
cleanup(sources): remove deprecated syntax 'source:' (singular source)

### PR DESCRIPTION
**Pull request by dduportal**
This PR removes the deprecated syntax `source:` (singular source).

Depends on #588 

Associated documentation PR: https://github.com/updatecli/website/pull/264

## Test

To test this pull request, you can run the following commands:

```shell
make test
```

### CODE REVIEW

File: pkg/core/config/main.go
Comment (lemeurherve) [inline pkg/core/config/main.go:146]:
```suggestion
				logrus.Errorf("The pullrequest %q specifies a scm id %q which does not exist", id, p.ScmID)
```

**Review by lemeurherve — DISMISSED** [review]

File: pkg/core/config/main.go
Comment (dduportal) [reply pkg/core/config/main.go]: resolved by rebasing from #588

**Review by lemeurherve — APPROVED** [review]

**Comment by dduportal** [comment]
- I'm waiting for review from 2 maintainers on this one (to be really sure that I did not do something wrong).
- Proposed milestone: 0.22.3

**Review by lemeurherve — DISMISSED** [review]

**Review by olblak — APPROVED** [review]
This cleanup is really great as it help my ongoing work about json schema generator

**Comment by dduportal** [comment]
Sorry, had to rebase due to merge conflicts. I need at least another new approval (putting this PR in auto-merge)

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**
